pennylane/plugins/tests/conftest.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains shared fixtures for the device tests."""
+import os
+import pytest
+import numpy as np
+import pennylane as qml
+
+# ==========================================================
+# pytest fixtures
+
+# seed for random functions
+np.random.seed(42)
+# Tolerance for analytic tests
+TOL = 1e-6
+# Tolerance for non-analytic tests
+TOL_STOCHASTIC = 0.05
+# Number of shots to call the devices with
+N_SHOTS = 50000
+# List of all devices that are included in PennyLane
+LIST_CORE_DEVICES = {"default.gaussian", "default.qubit", "default.qubit.tf"}
+
+
+@pytest.fixture(scope="function")
+def tol():
+    """Numerical tolerance for equality tests. Returns a different tolerance for tests
+    probing analytic or non-analytic devices (yielding deterministic or stochastic test results)."""
+
+    def _tol(analytic):
+        if analytic:
+            return float(os.environ.get("TOL", TOL))
+        else:
+            return TOL_STOCHASTIC
+
+    return _tol
+
+
+@pytest.fixture(scope="session")

```

Comment:
I know we want to keep fixtures at a minimum, but I find this one clutters the code if repeated in every function.